File: sources/academy/webscraping/scraping_basics_javascript/exercises/wta_tennis_links.mjs
import * as cheerio from 'cheerio';

const listingUrl = 'https://www.wtatennis.com/rankings/singles';
const response = await fetch(listingUrl);

if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

const html = await response.text();
const $ = cheerio.load(html);

for (const element of $('.rankings__list .player-row-drawer__link').toArray()) {
  const playerUrlRelative = $(element).attr('href');
  const playerUrl = new URL(playerUrlRelative, listingUrl).href;
  console.log(playerUrl);
}

File: sources/academy/webscraping/scraping_basics_python/exercises/wta_tennis_links.py
import httpx
from bs4 import BeautifulSoup
from urllib.parse import urljoin

listing_url = "https://www.wtatennis.com/rankings/singles"
response = httpx.get(listing_url)
response.raise_for_status()

soup = BeautifulSoup(response.text, "html.parser")

for link in soup.select(".rankings__list .player-row-drawer__link"):
    player_url = urljoin(listing_url, link["href"])
    print(player_url)
